test/test.js

```diff
@@ -310,6 +322,43 @@ describe("Test Screenshots", function() {
     }).catch(done);
   });
 
+  it("should download a shot", function(done) {
+    const startingFileCount = fs.readdirSync(downloadDir).length;
+    const filenameRegex = /^Screenshot.+ Firefox Screenshots\.png$/;
+    let files;
+    startAutoSelectionShot(driver)
+    .then(() => driver.wait(until.elementLocated(By.css(".highlight-button-download"))))
+    .then(downloadButton => downloadButton.click())
+    .then(() => driver.wait(() => {
+      return util.promisify(fs.readdir)(downloadDir).then(foundFiles => {
+        files = foundFiles;
+        return (files.length > startingFileCount);
+      });
+    }, 2000))
+    .then(() => {
+      // check for a file created within a couple seconds
+      // @TODO maybe check the dimensions of the image if we wanna get fancy
+      const matches = files.filter(filename => {
+        if (!filenameRegex.test(filename)) {
+          return false;
+        }
+        const stat = fs.statSync(path.join(downloadDir, filename));
+        return ((stat.mtime.getTime() - Date.now()) < 2000);
+      });
+      assert(matches.length, `Shot was not downloaded to ${downloadDir}`);
+      // clean up the dir
+      const unlinkPromise = util.promisify(fs.unlink);
+      files.forEach(filename => {
+        filename !== ".gitignore" && unlinkPromise(path.join(downloadDir, filename));
```

Comment:
Given that the promise isn't being used for anything, I guess promisifying isn't really needed here (though it doesn't hurt either). `return Promise.all(files.map(...))` would actually block on the cleanup, though I don't think that really matters either.